testnet/stacks-node/src/run_loop/neon.rs

```diff
@@ -31,6 +36,418 @@ pub struct RunLoop {
     coordinator_channels: Option<(CoordinatorReceivers, CoordinatorChannels)>,
 }
 
+/// Monitor the state of the Stacks blockchain as the peer network and relay threads download and
+/// proces Stacks blocks.  Don't allow the node to process the next PoX reward cycle's sortitions
+/// unless it's reasonably sure that it has processed all Stacks blocks for this reward cycle.
+/// This struct monitors the Stacks chainstate to make this determination.
+pub struct PoxSyncWatchdog {
```

Comment:
This struct should probably live somewhere like a `stacks-node::util` module -- it'll be useful to loops like krypton which would otherwise look pretty different from the neon run_loop.

Reply:
We don't have a krypton run loop -- it still uses the neon run loop.  If anything, we should be renaming neon-specific files to something more generic, since the same code powers neon, argon, krypton, and soon xenon.

Reply:
I still think this belongs in a different module than the outer run loop, which is ideally as small of a module as possible.

Reply:
Done